### PR TITLE
Fix hip-config.cmake for CMAKE_CXX_COMPILER=g++

### PR DESCRIPTION
**Pull request by yxsamliu**
Change-Id: I84ae83a5d223853706dd36834b7962ffe9573c1f

### CODE REVIEW

File: hip-config.cmake.in [hunk 1]
```diff
@@ -51,7 +51,9 @@ set_and_check(hip_HIPCC_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipcc")
 set_and_check(hip_HIPCONFIG_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipconfig")
 
 if(HIP_COMPILER STREQUAL "clang")
-  if(CMAKE_CXX_COMPILER MATCHES ".*hipcc")
+  if(HIP_CXX_COMPILER)
+    set(HIP_CLANG_ROOT ${HIP_CXX_COMPILER})
```
Comment (pfultz2) [inline hip-config.cmake.in:55]: This wont set the HIP_CLANG_ROOT correctly if `HIP_CXX_COMPILER` is set to /opt/rocm/llvm/bin/clang++.

**Comment by pfultz2** [comment]
It seems it would be better to do:

```cmake
if(NOT HIP_CXX_COMPILER)
set(HIP_CXX_COMPILER ${CMAKE_CXX_COMPILER})
endif()
```

And then do `s/CMAKE_CXX_COMPILER/HIP_CXX_COMPILER` for the rest of the file. This way if the user sets `HIP_CXX_COMPILER` to hipcc it can use the same logic to find the compiler.

**Comment by yxsamliu** [comment]
> It seems it would be better to do:
> 
> ```cmake
> if(NOT HIP_CXX_COMPILER)
> set(HIP_CXX_COMPILER ${CMAKE_CXX_COMPILER})
> endif()
> ```
> 
> And then do `s/CMAKE_CXX_COMPILER/HIP_CXX_COMPILER` for the rest of the file. This way if the user sets `HIP_CXX_COMPILER` to hipcc it can use the same logic to find the compiler.

Thanks. will do

**Comment by yxsamliu** [comment]
Revised.

File: hip-config.cmake.in [hunk 2]
```diff
@@ -60,9 +63,11 @@ if(HIP_COMPILER STREQUAL "clang")
     else()
       set(HIP_CLANG_ROOT /opt/rocm/llvm)
     endif()
-  else()
+  elseif (HIP_CXX_COMPILER MATCHES ".*clang++")
```
Comment (pfultz2) [inline hip-config.cmake.in:66]: This regex is not correct. Can we just use the `else` instead?

Comment (daniellowell) [reply hip-config.cmake.in:66]: `.*clang\\+\\+`

File: hip-config.cmake.in [hunk 3]
```diff
@@ -132,6 +137,12 @@ if(HIP_COMPILER STREQUAL "clang")
     INTERFACE_COMPILE_OPTIONS -x hip --hip-device-lib-path=${AMD_DEVICE_LIBS_PREFIX}/lib
   )
 
+  if (HIP_CXX_COMPILER MATCHES ".*clang\\+\\+")
+    set_property(TARGET hip::device APPEND PROPERTY
+      INTERFACE_COMPILE_OPTIONS -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false
+    )
```
Comment (atamazov) [inline hip-config.cmake.in:143]: [anchor]

**Pull request merged**